numpy/f2py/tests/test_crackfortran.py

```diff
@@ -45,6 +45,77 @@ def test_defaultPublic(self, tmp_path):
         assert "public" in mod["vars"]["seta"]["attrspec"]
 
 
+class TestModuleProcedure():
+    def test_defaultPrivate(self, tmp_path):
+        f_path = tmp_path / "mod.f90"
+        with f_path.open('w') as ff:
+            ff.write(textwrap.dedent("""\
+            module foo
```

Comment:
Very minor nit; we changed to a file based test interface in https://github.com/numpy/numpy/pull/20517.

Reply:
You're right. It's better with all this Fortran code in a file. Updated.